File: clojure/triangle/src/triangle.clj
(ns triangle)

(defn- triangle? [a b c] 
  (and (every? pos? [a b c]) (>= (+ a b) c) (>= (+ b c) a) (>= (+ a c) b)))

(defn equilateral?
  "Returns true if the triangle with sides a, b, and c is equilateral;
  otherwise, it returns false."
  [a b c]
  (and (triangle? a b c) (= a b c)))

(defn isosceles?
  "Returns true if the triangle with sides a, b, and c is isosceles;
  otherwise, it returns false."
  [a b c]
  (and (triangle? a b c) (>= 2 (count (set [a b c])))))

(defn scalene?
  "Returns true if the triangle with sides a, b, and c is scalene;
  otherwise, it returns false."
  [a b c]
  (and (triangle? a b c) (distinct? a b c)))

File: python/triangle/triangle.py
"""Functions to determine the type of a triangle"""

def equilateral(sides):
    return valid(sides) and len(set(sides)) == 1


def isosceles(sides):
    return valid(sides) and len(set(sides)) <= 2


def scalene(sides):
    return valid(sides) and len(set(sides)) == 3


def valid(sides):
    a, b, c = sides
    return (
        a > 0 and b > 0 and c > 0 and a + b > c and a + c > b and b + c > a
    )
    